pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # I'm folding summary, deprecation warning, and extended
```

Comment:
I don't really understand this comment. Summary isn't at the top right?

Reply:
For the numpydoc style, I'm assuming sections are separated by a line with a section name and a line of hyphens.  This means the summary (first line of docstring), deprecation warning (just after that, with no separator), and extended summary (everything from there to the first section separator) are treated together rather than separately.  This means I can't push the deprecation warning to the correct place or ensure proper spacing between the deprecation warning and extended summary.  It was also a lot easier to think about when I was writing.

Reply:
On a related note, what do you want `pydocstringformatter` to do if someone decides to include a `Preconditions` section (not in the standard) or `Prameters` (misspelled)?  I think the current sorting implementation produces an error, but I could probably change that to put all the weird ones at the end.